### PR TITLE
feat: [PL-29671]: Remove locking from health threads

### PR DESCRIPTION
**Pull request by bhavya181**
## Describe your changes
Instead of using synchronized for locking, we will use `ConcurrentHashmap` which blocks update operations of map for thread safety, this will not block whole map object.
`putIfAbsent` method of concurrenthashMap is used as it is atomic method, which are thread safe 

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40739)

### CODE REVIEW

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40739)* status: 1 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`980-commons/src/main/java/io/harness/health/HealthService.java` line 99 at r1](https://reviewable.io/reviews/harness/harness-core/40739#-NHy4TGN3Txf6nSlshdq:-NHy4TGN3Txf6nSlshdr:b-ok5vz3) ([raw file](https://github.com/harness/harness-core/blob/dc56b4cda3561473e187026e9e73ac258ce92dfe/980-commons/src/main/java/io/harness/health/HealthService.java#L99)):*
> ```Java
>   private void startCheck(HealthMonitor monitor) {
>     final Future<Throwable> future = executorService.submit(() -> checkMonitor(monitor));
>     monitors.replace(monitor,
> ```

why use replace?

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40739)* status: 1 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`980-commons/src/main/java/io/harness/health/HealthService.java` line 99 at r1](https://reviewable.io/reviews/harness/harness-core/40739#-NHy4TGN3Txf6nSlshdq:-NHyGPXJ0h6BbYcLWaMk:b-jcnn1h) ([raw file](https://github.com/harness/harness-core/blob/dc56b4cda3561473e187026e9e73ac258ce92dfe/980-commons/src/main/java/io/harness/health/HealthService.java#L99)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

why use replace?
</blockquote></details>

replace will prevent from adding unwanted key to map, but as keys will always come from map we are iterating, we can use put,
I have changed it

File: 980-commons/src/main/java/io/harness/health/HealthService.java [hunk 1]
```diff
@@ -50,18 +50,10 @@ static class Response {
     long validUntil;
   }
 
-  private Map<HealthMonitor, Response> monitors = new HashMap<>();
+  private Map<HealthMonitor, Response> monitors = new ConcurrentHashMap<>();
```
Comment (kshitij-harness) [inline 980-commons/src/main/java/io/harness/health/HealthService.java:53]: It's cleaner to use private ConcurrentMap<HealthMonitor, Response> monitors = new ConcurrentHashMap<>();

**Review by ankit-harness — DISMISSED** [review]
:lgtm:

> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40739#-:-NHySYWbE2xK2G-Aa92K:bnfp4nl)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Comment by bhavya181** [comment]
trigger feature-build

**Review by kshitij-harness — COMMENTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40739)* status: 1 of 2 approvals obtained.

**Review by kshitij-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40739)* status: :shipit: complete! ✓

**Review by kshitij-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40739)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by rijajoo — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40739)* status: :shipit: complete! ✓

**Pull request merged**